performance/gradle/benchmark.gradle

```diff
@@ -0,0 +1,130 @@
+apply plugin: 'kotlin-multiplatform'
+
+repositories {
+    maven {
+        url 'https://cache-redirector.jetbrains.com/jcenter'
+    }
+    maven {
+        url kotlinCompilerRepo
+    }
+    maven {
+        url buildKotlinCompilerRepo
+    }
+
+}
+
+
+private def determinePreset() {
+    def preset = MPPTools.defaultHostPreset(project)
+    println("$project has been configured for ${preset.name} platform.")
+    preset
+}
+
+def hostPreset = determinePreset()
+
+kotlin {
+    sourceSets {
+        commonMain {
+            dependencies {
+                implementation "org.jetbrains.kotlin:kotlin-stdlib-common:$kotlinVersion"
+            }
+            project.ext.commonSrcDirs.forEach {
+                kotlin.srcDir it
+            }
+
+        }
+        nativeMain {
+            project.ext.nativeSrcDirs.forEach {
+                kotlin.srcDir it
+            }
+        }
+        jvmMain {
+            dependencies {
+                implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk8:$kotlinVersion"
+            }
+            project.ext.jvmSrcDirs.forEach {
+                kotlin.srcDir it
+            }
+        }
+    }
+
+    targets {
+        fromPreset(presets.jvm, 'jvm') {
+            compilations.all {
+                tasks[compileKotlinTaskName].kotlinOptions {
+                    jvmTarget = '1.8'
+                }
+                tasks[compileKotlinTaskName].kotlinOptions.suppressWarnings = true
+            }
+        }
+
+        fromPreset(hostPreset, 'native') {
+            compilations.main.outputKinds('EXECUTABLE')
+            compilations.main.extraOpts '-opt'
+            compilations.main.buildTypes = [RELEASE]
+        }
+    }
+}
+
+MPPTools.addTimeListener(project)
+
+MPPTools.createRunTask(project, 'konanRun', kotlin.targets.native) {
+    workingDir = project.provider {
+        kotlin.targets.native.compilations.main.getBinary('EXECUTABLE', buildType).parentFile
+    }
+    args("-w", "$nativeWarmup", "-r", "$attempts", "-o", "${buildDir.absolutePath}/${nativeBenchResults}", "-p", "${project.ext.applicationName}::")
```

Comment:
What is `::` here?

Reply:
It's just end of prefix to have something like "Ring::ClassArray.allocateArray" in report.